eZ/Publish/API/Repository/Values/ContentType/ContentType.php

```diff
@@ -184,7 +184,49 @@ abstract class ContentType extends ValueObject implements MultiLanguageName, Mul
      *
      * @param string $fieldDefinitionIdentifier
      *
-     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition|null
      */
     abstract public function getFieldDefinition($fieldDefinitionIdentifier);
+
+    /**
+     * This method returns true if the field definition for the given identifier exists.
+     *
+     * @param string $fieldDefinitionIdentifier
+     *
+     * @return bool
+     */
+    public function hasFieldDefinition(string $fieldDefinitionIdentifier): bool
+    {
+        return $this->getFieldDefinition($fieldDefinitionIdentifier) !== null;
+    }
+
+    /**
+     * Returns first field definition with given field type identifier or null.
+     *
+     * @param string $fieldTypeIdentifier
+     *
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition
+     */
+    public function getFieldDefinitionWithType(string $fieldTypeIdentifier): ?FieldDefinition
```

Comment:
'ofType' instead 'withType' ?

Reply:
also, might add 2nd param `int $offset = 0`. This way we could get the 1st image filed, the 2nd one, etc...

Reply:
Do we need `getFirstFieldDefinitionWithType()` and `getFieldDefinitionsWithType()` ? My question about the use-case for this still stands :)

Reply:
@bdunogier if you have the getFieldDefinitionsWithType($limit, $offset), I don't think both the getFirstFieldDenfinitionWithType() or getFieldDefinitionWithType()  method would be needed at all.